core/src/main/scala/org/apache/spark/util/AccumulatorV2.scala

```diff
@@ -268,7 +268,7 @@ private[spark] object AccumulatorContext extends Logging {
       // Since we are storing weak references, warn when the underlying data is not valid.
       val acc = ref.get
       if (acc eq null) {
-        logWarning(s"Attempted to access garbage collected accumulator $id")
+        logInfo(s"Attempted to access garbage collected accumulator $id")
```

Comment:
In addition to `Jenkins`, `spark-shell` (and users) also will lose this warning because the default level is `WARN`.

cc @cloud-fan since this is designed by him.

Reply:
At first, I thought this won't happen frequently. Maybe I was wrong. @ericl can you give more details here? e.g. why it happens so frequently in the tests.

Reply:
I wouldn't mind writing `acc == null` in the line above too, but whatever, not related.

Reply:
I see this actually doesn't happen that often in OSS jenkins, so reverting this.